demisto_sdk/commands/common/hook_validations/integration.py

```diff
@@ -599,6 +617,96 @@ def _get_invalid_dbot_outputs(context_outputs_paths, context_outputs_description
 
         return missing_outputs, missing_descriptions
 
+    def validate_all_reputation_outputs_exist(
+        self, command: dict, output_results: OutputsResults
+    ) -> bool:
+        """check if all mandatory outputs for custom object do exist.
+        Args:
+            command_name: the command name
+            outputs: outputs of the command
+            used_reputation_objects: reputation objects (like url, ip, etc) used in the command's context outputs.
+        Returns:
+            bool: Whether all mandatory outputs of custom outputs exist or not
+        """
+        required_outputs: Set[str] = set()
+        for ioc in output_results.reputation_objects:
+            required_outputs.update(REPUTATION_TO_REQUIRED_OUTPUT.get(ioc.lower(), ()))
+
+        missing_outputs = required_outputs - set(output_results.context_outputs_paths)
+
+        for output in missing_outputs:
+            error_message, error_code = Errors.command_reputation_output_is_missing(
+                command.get("name", ""),
+                output,
+                ioc,
+            )
+            if self.handle_error(
+                error_message,
+                error_code,
+                file_path=self.file_path,
+                warning=self.structure_validator.quiet_bc,
+            ):
+                return False
+
+        return True
+
+    @staticmethod
+    def extract_object_name(context_path):
+        for object_name in MANDATORY_REPUTATION_CONTEXT_OBJECTS_NAMES:
+            if object_name.lower() in context_path.lower():
+                return object_name
+        return None
```

Comment:
1. it's only used in one place, and not very complex, so I doubt we need it as a method
2. name doesn't really say what it does
3. not documented

let's delete the method, copy the code into where it's used (or make it an internal method)